org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/tool/EclipseFileManager.java

```diff
@@ -151,13 +152,14 @@ private void collectAllMatchingFiles(Location location, File file, String normal
 			Archive archive = this.getArchive(file);
 			if (archive != Archive.UNKNOWN_ARCHIVE) {
 				String key = normalizedPackageName;
+				boolean findAll = key.length() == 0;
 				if (!normalizedPackageName.endsWith("/")) {//$NON-NLS-1$
 					key += '/';
 				}
 				// we have an archive file
 				if (recurse) {
 					for (String packageName : archive.allPackages()) {
-						if (packageName.startsWith(key)) {
+						if (findAll || packageName.startsWith(key)) {
```

Comment:
if `findAll` is true, the `key` will have zero length... looks like `.startsWith` will still return true in that case

```
jshell> "foo".startsWith("")
$1 ==> true
```
is the `findAll` check needed?

Reply:
Thanks for checking, but key will never have length 0, see `key += '/'` just above.